00X-OFEP-provider-flag-metadata.md

```diff
@@ -0,0 +1,50 @@
+## Surfacing flag metadata
+
+## State: DRAFTING
+This proposal lays out a mechanism for flag providers to surface arbitrary flag metadata to Open Feature, and for hooks to access this metadata.
+
+
+## Background
+
+Flag providers maintain metadata about a feature flag, and there are scenarios where a hook might benefit from being able to access this metadata. For example, a flag provider could surface a `management-url` attribute which an open-telemetry hook could then add to an otel span, allowing someone viewing a trace for a feature-flagged operation to easily navigate to a webpage describing that flag in detail.
+
+Different providers will expose different kinds of metadata, and different Application Integrators will want to consume that metadata in different ways. As such, the goal of this proposal is to provide a simple way for the Open Feature runtime to pass generic metadata from a provider to a hook, without Open Feature itself understanding the semantics of that metadata.
+
+## Non-goals
+
+In the future we may wish to standardize the semantics of some key metadata attributes which are common amongst many providers (e.g. `expiration-date`, `owner`, `management-url`) so that provider-agnostic hooks can be created to consume these standard attributes, but that is explicitly NOT in the scope of this OFEP - we would prefer to see which common attributes naturally emerge and then "pave those cow paths".
+
+## Proposal
+
+The Open Feature spec will be extended in two ways:
+- a flag provider will be able to provide a bag of metadata attributes for a feature flag. This bag will be a set of key-value pairs, where the key will be a string and the value will be a small set of primitive values.
+- a hook will be able to access these metadata attributes
+
+*note* what follows is a VERY rough strawman proposal for one way to extend the spec. Please poke holes!
+
+### provider interface
+
+We add a `getMetadataForFlag(flagKey:string):FlagMetadata` method to the provider interface. 
```

Comment:
I would expect this to come back in the `ProviderEvaluation` object or equivalent?

https://github.com/open-feature/java-sdk/blob/main/src/main/java/dev/openfeature/sdk/FeatureProvider.java

Reply:
Wow, not sure why it didn't occur to me to put it there instead - that makes much more sense!

Reply:
@justinabrahms updated the proposal per your suggestion: ace0ba9abe534ec064076749078353e6f791a310